components/camel-kudu/src/main/java/org/apache/camel/component/kudu/KuduProducer.java

```diff
@@ -0,0 +1,112 @@
+import org.apache.kudu.client.Insert;
+import org.apache.kudu.client.KuduClient;
+import org.apache.kudu.client.KuduException;
+import org.apache.kudu.client.KuduTable;
+import org.apache.kudu.client.PartialRow;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The Kudu producer.
+ *
+ * @see org.apache.camel.component.kudu.KuduEndpoint
+ */
+public class KuduProducer extends DefaultProducer {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KuduProducer.class);
+
+    private final KuduEndpoint endpoint;
+
+    public KuduProducer(KuduEndpoint endpoint) {
+        super(endpoint);
+
+        if (endpoint == null || endpoint.getKuduClient() == null) {
+            throw new IllegalArgumentException("Can't create a producer when the database connection is null");
+        }
+        this.endpoint = endpoint;
+    }
+
+    @Override
+    public void process(Exchange exchange) throws Exception {
+        String table = endpoint.getTableName();
+        switch (endpoint.getOperation()) {
+        case INSERT:
+            doInsert(exchange, table);
+            break;
+        case CREATE_TABLE:
+            doCreateTable(exchange, table);
+            break;
+        case SCAN:
+            doScan(exchange, table);
+            break;
+        default:
+            throw new IllegalArgumentException("The operation " + endpoint.getOperation() + " is not supported");
+        }
+    }
+
+    private void doInsert(Exchange exchange, String tableName) throws KuduException {
+        LOG.trace("Insert on table {}", tableName);
+        KuduClient connection = endpoint.getKuduClient();
+        KuduTable table = connection.openTable(tableName);
+
+
+        Insert insert = table.newInsert();
+        PartialRow row = insert.getRow();
+
+        Map<?, ?> rows = exchange.getIn().getBody(Map.class);
+        for (Map.Entry<?, ?> entry : rows.entrySet()) {
+            final String colName = entry.getKey().toString();
+            final Object value = entry.getValue();
+            //Add only if column exist
+            //If not, this will throw an IllegalArgumentException
+            if (table.getSchema().getColumn(colName) != null) {
+                row.addObject(colName, value);
+            }
+        }
+
+        connection.newSession().apply(insert);
+    }
+
+    private void doCreateTable(Exchange exchange, String tableName) throws KuduException {
+        LOG.trace("Creating table {}", tableName);
+        KuduClient connection = endpoint.getKuduClient();
+
+        try {
+            Schema schema = (Schema) exchange.getIn().getHeader(KuduConstants.CAMEL_KUDU_SCHEMA);
+            CreateTableOptions builder = (CreateTableOptions)
+                                             exchange.getIn()
+                                                 .getHeader(KuduConstants.CAMEL_KUDU_TABLE_OPTIONS);
+            connection.createTable(tableName, schema, builder);
+        } catch (Throwable t) {
```

Comment:
Since you are added checked exception `KuduException`, you don't need to catch and re-throw it again

Reply:
This was for debugging, shouldn't be here :(